File: spring-rest-configuration-autodiscover.scala/src/main/scala/demo/service/core/PropertyBooleanService.scala
package demo.service.core

import java.util._

abstract class PropertyBooleanService extends AbstractPropertyService[Boolean] {

  override def getType: Type.Value = Type.BOOLEAN

  override def fromString(input: String): Boolean =
    if (input == null) getDefaultValue else java.lang.Boolean.valueOf(input)

  override def toString(input: Boolean): String =
    if (input == null) getDefaultValue.toString else String.valueOf(input)

  override def getPossibleValues: java.util.List[String] = java.util.Arrays.asList("true", "false")
}

File: spring-rest-configuration-autodiscover.java/src/main/java/demo/service/core/PropertyBooleanService.java
package demo.service.core;

import java.util.Arrays;
import java.util.List;

public abstract class PropertyBooleanService extends AbstractPropertyService<Boolean> {

    @Override
    public Type getType() {
        return Type.BOOLEAN;
    }

    @Override
    public Boolean fromString(String input) {
        return input == null ? getDefaultValue() : Boolean.valueOf(input);
    }

    @Override
    public String toString(Boolean input) {
        return (input == null) ? getDefaultValue().toString() : String.valueOf(input);
    }

    public List<String> getPossibleValues() {
        return Arrays.asList("true", "false");
    }
}
